sdk/how-to-guides/manage-storage.md

```diff
@@ -0,0 +1,232 @@
+
+```python
+@pytest.fixture()
+def harness():
+    yield ops.testing.Harness(MyCharm)
+    harness.cleanup()
+
+
+def test_storage_attached(harness):
+    # Add one instance of the expected storage to the charm. This is before `.begin()` is called,
+    # so will not trigger any events.
+    storage_id = harness.add_storage("cache", 1)
+    harness.begin()
+    # Simulate Juju attaching the storage, which will trigger a storage-attached event on the charm.
+    harness.attach_storage(storage_id)
+    # Assert that it was handled correctly.
+    assert ...
+
+
+def test_storage_detaching(harness):
+    storage_id = harness.add_storage("cache", 1, attach=True)
+    harness.begin()
+    # Simulate the harness being detached (.remove_storage() would simulate it being removed
+    # entirely).
+    harness.remove_storage(storage_id)
+    # Asser that it was handled correctly.
+    assert ...
+```
+
+> See more: [`ops.testing.Harness.add_storage`](https://ops.readthedocs.io/en/latest/#ops.testing.Harness.add_storage), [`ops.testing.Harness.attach_storage`](https://ops.readthedocs.io/en/latest/#ops.testing.Harness.attach_storage), [`ops.testing.Harness.detach_storage`](https://ops.readthedocs.io/en/latest/#ops.testing.Harness.detach_storage), [`ops.testing.harness.remove_storage`](https://ops.readthedocs.io/en/latest/#ops.testing.Harness.remove_storage) 
+
+<a href="#heading--write-scenario-tests"><h3 id="heading--write-scenario-tests">Write scenario tests</h3></a>
+
+> See first: [How to write scenario tests for a charm](/t/10585)
+
+When using Scenario for unit tests, to verify that the charm state is as expected after storage changes, use the `run` method of the Scenario `Context` object. For example, to provide the charm with mock storage:
+
+```python
+# Some charm with a 'foo' filesystem-type storage defined in its metadata:
+ctx = scenario.Context(MyCharm)
+storage = scenario.Storage("foo")
+
+# Setup storage with some content:
+(storage.get_filesystem(ctx) / "myfile.txt").write_text("helloworld")
+
+with ctx.manager("update-status", scenario.State(storage=[storage])) as mgr:
+    foo = mgr.charm.model.storages["foo"][0]
+    loc = foo.location
+    path = loc / "myfile.txt"
+    assert path.exists()
+    assert path.read_text() == "helloworld"
+
+    myfile = loc / "path.py"
+    myfile.write_text("helloworlds")
+
+# Verify that the contents are as expected afterwards.
+assert (
+    storage.get_filesystem(ctx) / "path.py"
+).read_text() == "helloworlds"
+```
+
+If a charm requests adding more storage instances while handling some event, you
+can inspect that from the Context.requested_storage API.
+
+```python
+ctx = scenario.Context(MyCharm)
+ctx.run('some-event-that-will-request-more-storage', scenario.State())
+
+# The charm has requested two 'foo' storages to be provisioned:
+assert ctx.requested_storages['foo'] == 2
+```
+
+Requesting storages has no other consequence in Scenario. In real life, this
```

Comment:
Is "storages" okay? I've never heard it in the plural. I'm guessing it's understood as storage volumes, but even then -- is it common enough to use this way?

Reply:
Covered under the "storage" question above - I'll adjust all the cases to match.